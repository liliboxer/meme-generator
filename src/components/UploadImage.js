import React from 'react';
import PropTypes from 'prop-types';

function UploadImage({ handleUpload, handleChange, imageURL  }) {
  return (
    <section>
      <input type="file" onChange={handleUpload}/>
      <input name="imageURL" value={imageURL}placeholder="image URL" onChange={(event) => handleChange(event)}></input>
    </section>
  );
}

UploadImage.propTypes = {
  handleUpload: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  imageURL: PropTypes.string.isRequired
};

export default UploadImage;
