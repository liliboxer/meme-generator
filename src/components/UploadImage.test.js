import React from 'react';
import { shallow } from 'enzyme';
import UploadImage from './UploadImage';

describe('UploadImage component', () => {
  it('renders UploadImage', () => {
    const wrapper = shallow(<UploadImage />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
