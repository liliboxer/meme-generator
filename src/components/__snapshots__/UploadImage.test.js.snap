// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadImage component renders UploadImage 1`] = `
<section>
  <input
    type="file"
  />
  <input
    name="imageURL"
    onChange={[Function]}
    placeholder="image URL"
  />
</section>
`;
